sdl_android/src/main/java/com/smartdevicelink/streaming/RTPH264Packetizer.java

```diff
@@ -0,0 +1,488 @@
+	private final static int TLS_MAX_RECORD_SIZE = 16384;
+	private final static int TLS_RECORD_HEADER_SIZE = 5;
+	private final static int TLS_RECORD_MES_AUTH_CDE_SIZE = 32;
+	private final static int TLS_MAX_RECORD_PADDING_SIZE = 256;
+
+	private final static int MAX_DATA_SIZE_FOR_ENCRYPTED_SERVICE =
+			TLS_MAX_RECORD_SIZE - TLS_RECORD_HEADER_SIZE - TLS_RECORD_MES_AUTH_CDE_SIZE- TLS_MAX_RECORD_PADDING_SIZE;
+
+	private boolean mServiceProtected;
+	private Thread mThread;
+	private BlockingQueue<ByteBuffer> mOutputQueue;
+	private volatile boolean mPaused;
+	private boolean mWaitForIDR;
+	private NALUnitReader mNALUnitReader;
+	private byte mPayloadType = 0;
+	private int mSSRC = 0;
+	private char mSequenceNum = 0;
+	private int mInitialPTS = 0;
+
+	/**
+	 * Constructor
+	 *
+	 * @param streamListener The listener which this packetizer outputs SDL frames to
+	 * @param serviceType The value of "Service Type" field in SDL frames
+	 * @param sessionID The value of "Session ID" field in SDL frames
+	 * @param session The SdlSession instance that this packetizer belongs to
+	 */
+	public RTPH264Packetizer(IStreamListener streamListener,
+	                         SessionType serviceType, byte sessionID, SdlSession session) throws IOException {
+
+		super(streamListener, null, serviceType, sessionID, session);
+
+		mServiceProtected = session.isServiceProtected(_serviceType);
+
+		if (bufferSize == 0) {
+			// fail safe
+			bufferSize = MAX_DATA_SIZE_FOR_ENCRYPTED_SERVICE;
+		}
+		if (mServiceProtected && bufferSize > MAX_DATA_SIZE_FOR_ENCRYPTED_SERVICE) {
+			bufferSize = MAX_DATA_SIZE_FOR_ENCRYPTED_SERVICE;
+		}
+
+		mOutputQueue = new LinkedBlockingQueue<ByteBuffer>(MAX_QUEUE_SIZE / bufferSize);
+		mNALUnitReader = new NALUnitReader();
+		mPayloadType = DEFAULT_RTP_PAYLOAD_TYPE;
+
+		Random r = new Random();
+		mSSRC = r.nextInt();
+
+		// initial value of the sequence number and timestamp should be random ([5.1] in RFC3550)
+		mSequenceNum = (char)r.nextInt(65536);
+		mInitialPTS = r.nextInt();
+	}
+
+	/**
+	 * Sets the Payload Type (PT) of RTP header field.
+	 *
+	 * Use this method if PT needs to be specified. The value should be between 0 and 127.
+	 * Otherwise, a default value (96) is used.
+	 *
+	 * @param type A value indicating the Payload Type
+	 */
+	public void setPayloadType(byte type) {
+		if (type >= 0 && type <= 127) {
+			mPayloadType = type;
+		} else {
+			mPayloadType = DEFAULT_RTP_PAYLOAD_TYPE;
+		}
+	}
+
+	/**
+	 * Sets the SSRC of RTP header field.
+	 *
+	 * Use this method if SSRC needs to be specified. Otherwise, a random value is generated and
+	 * used.
+	 *
+	 * @param ssrc An integer value representing SSRC
+	 */
+	public void setSSRC(int ssrc) {
+		mSSRC = ssrc;
+	}
+
+	/**
+	 * Overwrites MTU value. This is only for testing.
+	 *
+	 * @param mtu maximum payload size of SDL frames
+	 */
+	public void setMTU(int mtu) {
+		if (mtu <= 0) {
+			mtu = MAX_DATA_SIZE_FOR_ENCRYPTED_SERVICE;
```

Comment:
See comment above. Also should check if the encryption flag is set before setting MTU.

Reply:
Actually this method was intended for testing. I found that private member variable can be altered with reflection, so I've updated the test case and removed this method completely with commit https://github.com/smartdevicelink/sdl_android/pull/592/commits/5325fc61689d6022006ab250b43096934a78750c. Sorry for your confusion!